src/treelearner/feature_histogram.hpp

```diff
@@ -582,6 +670,8 @@ class FeatureHistogram {
         t = -1;
       }
 
+      constraints.InitializeIndices(dir);
```

Comment:
`constraints` is not the reay-only object?

Reply:
could we pass `dir` to  `CurrentMinConstraintRight` (and other similar function), to make `constraints` read-only?

Reply:
Up on this in case you didn't see it @guolinke.